File: src/problem004/solution004.c
#include <stdio.h>
#include <math.h>

isPalindrome(int n)
{
  int rev = 0, mod = n;
  while(mod){
    rev = 10 * rev + (mod % 10);
    mod /= 10;
  }
  return rev == n;
}

solve(){
    int n = 999, i, j;
    for(i=0;i<2*n+1;++i){
        for(j=0;j<i+1;j++){
            if((j + i) % 2 == 0){
                int a = (int)(n - (i - j - 1)/2.0);
                int pal = a * (a - j);
                if(isPalindrome(pal))
                    return pal;
            }
        }
    }
}

main(){
    printf("%d\n", solve());
}

File: src/problem004/solution004.py
def solve():
    n = 999
    for i in xrange(2 * n + 1):
        for j in (x for x in xrange(i + 1) if (x + i) % 2 == 0):
            a = int(n - (i - j - 1)/2.0)
            pal = str(a * (a - j))
            if pal == pal[::-1]:
                return pal

if __name__ == '__main__':
    print(solve())